### ISSUE TITLE
Generate UUID token on the backend to prevent unauthorized access to manually started backend processes

### ISSUE THREAD

**Issue by Kechil** [CARTAvis/carta-backend#676]
This issue deals with an important change required in order to make the execution of CARTA (outside of the carta-node-server case) more secure and more familiar to astronomers. The approach is based on that of [jupyter-notebook](https://jupyter-notebook.readthedocs.io/en/stable/security.html).

The backend currently accepts a pre-generated token via the `CARTA_AUTH_TOKEN` environmental variable. If this is missing, the backend will accept any and all incoming websocket connections. This presents a security risk when running in what is currently known as "remote mode", where a backend is started on a server and accessed by the user in a browser.

Instead we should generate a UUID token (random-based, rather than time or hardware address based) if no token is passed in through the `CARTA_AUTH_TOKEN` environmental variable. 

This feature, along with #670 and #671 will allow us to print out a custom URL for the user to connect to as soon as the backend is started. For reference, this is what jupyter-notebook does:

```
angus@angus-ryzen:~$ jupyter-notebook 
[I 14:32:54.017 NotebookApp] Serving notebooks from local directory: /home/angus
[I 14:32:54.017 NotebookApp] The Jupyter Notebook is running at:
[I 14:32:54.017 NotebookApp] http://localhost:8888/?token=2be16b2a306f1d9288b6c1fd9153357d5a0dc229117bb19d
[I 14:32:54.017 NotebookApp]  or http://127.0.0.1:8888/?token=2be16b2a306f1d9288b6c1fd9153357d5a0dc229117bb19d
[I 14:32:54.017 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

This presensts the user with a simple and (more) secure way of running CARTA without relying on "remote mode" tacked on to the electron app.

The backend should also have a `--no-token` command-line argument that can be used to prevent the token from being generated, for debug/development purposes.

**Comment by robsimmonds** [comment]
Where will this be generated? If it is done at user level it will be a lot of effort to make it secure. I feel this is all still going in the wrong direction.

**Comment by veggiesaurus** [comment]
> Where will this be generated? If it is done at user level it will be a lot of effort to make it secure.

I propose using the linux-standard `libuuid` library (specifically [uuid_generate_random](https://linux.die.net/man/3/uuid_generate_random)) to generate this.

I agree that utilising the carta-node-server would be more secure, but in my mind, this is a step in the right direction for the _other_ use cases, which will not be entirely covered by carta-node-server just yet. This approach is definitely an improvement over just running `carta --remote`, which would allow _any_ user the ability to connect to that carta-backend process with no security measures whatsoever.

**Comment by veggiesaurus** [comment]
in progress, will be merged in at the same time as #670
